### PR TITLE
Dev 2865 use hook in publish modal

### PR DESCRIPTION
**Pull request by nrh-cklimas**
Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)

1. https://github.com/newsrevenuehub/rev-engine/pull/843
2. https://github.com/newsrevenuehub/rev-engine/pull/882 (this PR)
3. https://news-revenue-hub.atlassian.net/browse/DEV-2866
4. https://news-revenue-hub.atlassian.net/browse/DEV-2867

#### What's this PR do?

- Changes PublishButton so it uses the `useContributionPage` hook created in DEV-2865 instead of making an API request itself.
- Changes `useContributionPage` so that it can take either a RP slug + page slug as arguments OR page ID. I think most cases will use the slug invocation, but the publish button specifically needs to use ID so that if the user changes the slug on publish, it doesn't get confused about what page it's working on.
- Removes the RETooltip component in favor of our new base tooltip.

#### Why are we doing this? How does it help us?

Simplifies our component code so API interactions occur in one well-tested place.

#### How should this be manually tested? Please include detailed step-by-step instructions.

1. Edit an unpublished page. Confirm that you are able to publish it and that the slug you enter in the publish modal is saved.
2. Click the "Published" button on a published page and confirm that the usual popover showing its URL appears.

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?

No.

#### Have automated unit tests been added? If not, why?

Yes.

#### How should this change be communicated to end users?

n/a

#### Are there any smells or added technical debt to note?

No.

#### Has this been documented? If so, where?

No.

#### What are the relevant tickets? Add a link to any relevant ones.

https://news-revenue-hub.atlassian.net/browse/DEV-2865

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?

No.

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:

See other stacked PRs linked above.

### CODE REVIEW

File: spa/src/components/common/Button/PublishButton/PublishButton.tsx [hunk 1]
```diff
@@ -8,117 +8,142 @@ import FiberManualRecordIcon from '@material-ui/icons/FiberManualRecord';
...
-        onClose={handleCloseTooltip}
-        onOpen={handleOpenTooltip}
       >
         {/* Disabled elements do not fire events. Need the DIV over button for tooltip to listen to events. */}
         <div data-testid="publish-button-wrapper">
           <Button
             variant="contained"
             onClick={isPublished ? handleOpenPopover : handleOpen}
-            active={showPopover ? 'true' : ''}
+            $active={showPopover}
```
Comment (GuiMend) [inline spa/src/components/common/Button/PublishButton/PublishButton.tsx:131]: How did you come up with this? Very interesting and I have never seen before.

Also, does this remove the console errors when running jest?

File: spa/src/hooks/useContributionPage/useContributionPage.ts [hunk 2]
```diff
@@ -59,7 +65,10 @@ export interface UseContributionPageResult {
   ) => Promise<void>;
 }
 
-export function useContributionPage(revenueProgramSlug: string, pageSlug: string): UseContributionPageResult {
+export function useContributionPage(pageId?: number): UseContributionPageResult;
+export function useContributionPage(revenueProgramSlug: string, pageSlug: string): UseContributionPageResult;
+
+export function useContributionPage(revenueProgramSlugOrPageId?: number | string, pageSlug?: string) {
```
Comment (GuiMend) [inline spa/src/hooks/useContributionPage/useContributionPage.ts:71]: This is also awesome. How does this work? Having multiple declararions of the same function 🤔

**Review by GuiMend — COMMENTED** [review]
Nice job, learned some new thing reviewing this PR 👍

File: spa/src/components/common/Button/PublishButton/PublishButton.stories.tsx [hunk 3]
```diff
@@ -1,4 +1,7 @@
-import PublishButton from './PublishButton';
+import { ComponentMeta, ComponentStory } from '@storybook/react';
+import PublishButton, { PublishButtonProps } from './PublishButton';
+
+// need to wrap with query client provider
```
[the remark this answers is not in the document]
Comment (nrh-cklimas) [reply spa/src/components/common/Button/PublishButton/PublishButton.stories.tsx:4]: Oops... meant to address this before I opened the PR. Fixed in 4da939ed120c7fdf13199ea0fb24e80b7e070f87

File: spa/src/components/common/Button/PublishButton/PublishButton.styled.ts [hunk 4]
```diff
@@ -8,7 +8,7 @@ export const Flex = styled.div`
   font-family: ${(props) => props.theme.systemFont};
 `;
 
-export const Button = styled(MuiButton)`
+export const Button = styled(MuiButton)<{ $active?: boolean; $published?: boolean }>`
```
[the remark this answers is not in the document]
Comment (nrh-cklimas) [reply spa/src/components/common/Button/PublishButton/PublishButton.styled.ts:11]: I think we're going to need to use these as we convert more of the app to TypeScript, in fact. Styled Components uses the dollar sign as a convention to signal it's a [transient prop](https://styled-components.com/docs/api#transient-props) that only Styled Components sees, and isn't passed onto the underlying component. I got this tip from [their TypeScript docs](https://styled-components.com/docs/api#using-custom-props) which talks about it more.

File: spa/src/components/common/Button/PublishButton/PublishButton.tsx (hunk 1 above)
Comment (nrh-cklimas) [reply spa/src/components/common/Button/PublishButton/PublishButton.tsx:131]: See my note about transient props in the style definition. Not sure about the Jest console errors--depends on which ones you're referring to? I still seem warnings on my Jest runs.

File: spa/src/hooks/useContributionPage/useContributionPage.ts (hunk 2 above)
Comment (nrh-cklimas) [reply spa/src/hooks/useContributionPage/useContributionPage.ts:71]: Yeah, it's really useful in TypeScript! [Here are docs on it](https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads).

File: spa/src/components/common/Button/PublishButton/PublishButton.styled.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (GuiMend) [reply spa/src/components/common/Button/PublishButton/PublishButton.styled.ts:11]: Wow, that's awesome. Didn't know about them before. 👏

**Review by GuiMend — APPROVED** [review]
👍

**Pull request merged**